app/controllers/hmis_controller.rb

```diff
@@ -0,0 +1,58 @@
+
+class HmisController < ApplicationController
+  # TODO: this is a temporary proxy for access
+  before_action :require_can_upload_hud_zips!
+  before_action :set_item, only: [:show]
+  before_action :searched?, only: [:index]
+
+  include ArelHelper
+
+  def index
+    if searched?
+      @type = params[:search].try(:[], :type)
+      @id = params[:search].try(:[], :id)
+    end
+
+    @results = load_results
+
+  end
+
+  def show
+
+  end
+
+  private def searched?
+    @searched = params[:search].present?
+  end
+  helper_method :searched?
+
+  private def load_results
+    return [] unless @searched
+    @klass = valid_class(params[:search].try(:[], :type))
```

Comment:
The previous line renders the try unnecessary... Remove it, or for safety, add the try to the id test? Or alternatively, if you move the call to `load_results` into the `if` both the try and the previous line become unnecessary, yes?

Reply:
I believe the try is necessary because you might have `params[:search][:foo]` but no `:type`.  The pass to `valid_class` would throw an error if there were no try.